sqlglot/expressions.py

```diff
@@ -3554,8 +3581,22 @@ def build(
 
         return DataType(**{**data_type_exp.args, **kwargs})
 
-    def is_type(self, *dtypes: str | DataType | DataType.Type) -> bool:
-        return any(self.this == DataType.build(dtype).this for dtype in dtypes)
+    def is_type(self, *dtypes: str | DataType | DataType.Type, only_kind: bool = False) -> bool:
+        """
+        Checks whether this DataType matches one of the provided data types.
+
+        Args:
+            dtypes: the data types to compare this DataType to.
```

Comment:
i feel like by default it should be something like

array == array
array[int] = array
array[int] != array[double]

float = float
float = float(10.0)
float(1.0) != float(10.0)

does that make sense?

Reply:
yeah I was a bit skeptical regarding the type comparisons.. i'm not sure if allowing `type = type(precision)` is correct, e.g. `varchar` is not necessarily the same as `varchar(10)`

Reply:
if you're asking for a varchar, you don't really care if it's a varchar or a varchar10,

same if you're asking if something is an array, you don't care if it's an array<str> or array<int>

Reply:
updated, but might need a little more refactoring - let me know what you think